examples/pxScene2d/src/jsbindings/rtWrapperUtils.cpp

```diff
@@ -111,36 +111,42 @@ void weakCallback_rt2v8(const WeakCallbackData<Object, rtIObject>& data)
 void
 HandleMap::clearAllForContext(uint32_t contextId)
 {
-  // JRJR Bail for now a bit crashy
-  return;
-
   typedef ObjectReferenceMap::iterator iterator;
 
   int n = 0;
 
   pthread_mutex_lock(&sObjectMapMutex);
   rtLogInfo("clearing all persistent handles for: %u size:%u", contextId,
     static_cast<unsigned>(objectMap.size()));
-
+  vector<iterator> refs;
   for (iterator begin = objectMap.begin(), end = objectMap.end(); begin != end;)
   {
-    ObjectReference* ref = begin->second;
-    // rtLogInfo("looking at:%d == %d", ref->CreationContextId, contextId);
+      ObjectReference* ref = begin->second;
+      rtLogInfo("looking at:%d == %d", ref->CreationContextId, contextId);
 
-    if (ref->CreationContextId == contextId)
-    {
-      ref->PersistentObject.ClearWeak();
-      ref->PersistentObject.Reset();
-      delete ref;
-      objectMap.erase(begin++);
```

Comment:
Can we do: begin = objectMap.erase(begin); 

instead of adding a new vector<iterator> refs?